azkaban-common/src/main/java/azkaban/utils/OsCpuUtil.java

```diff
@@ -0,0 +1,179 @@
+
+  /**
+   * Collects a new cpu stat data point and calculates cpu load with it and the oldest one
+   * collected which is then deleted.
+   *
+   * @return percentage of CPU usage. -1 if there are no cpu stats.
+   */
+  public double getCpuLoad() {
+    if (this.collectedCpuStats.isEmpty()) {
+      return -1;
+    }
+    final CpuStats oldestCpuStats = this.collectedCpuStats.pollLast();
+    final CpuStats newestCpuStats = getCpuStats();
+    this.collectedCpuStats.push(newestCpuStats);
+
+    return calcCpuLoad(oldestCpuStats, newestCpuStats);
+  }
+
+  private double calcCpuLoad(final CpuStats startCpuStats, final CpuStats endCpuStats) {
+    final long startSysUptime = startCpuStats.getSysUptime();
+    final long startTimeCpuIdle = startCpuStats.getTimeCpuIdle();
+    final long endSysUptime = endCpuStats.getSysUptime();
+    final long endTimeCpuIdle = endCpuStats.getTimeCpuIdle();
+
+    if (endSysUptime == startSysUptime) {
+      logger.error("Failed to calculate cpu load: division by zero");
+      return -1.0;
+    }
+    final double percentageCpuIdle =
+        (100.0 * (endTimeCpuIdle - startTimeCpuIdle)) / (endSysUptime - startSysUptime);
+    return 100.0 - percentageCpuIdle;
+  }
+
+  private CpuStats getCpuStats() {
+    if (!Files.isRegularFile(Paths.get(CPU_STAT_FILE))) {
+      // Mac doesn't use proc pseudo files for example.
+      return null;
+    }
+
+    final String cpuLine = getCpuLineFromStatFile();
+    if (cpuLine == null) {
+      return null;
+    }
+
+    return getCpuStatsFromLine(cpuLine);
+  }
+
+  private String getCpuLineFromStatFile() {
+    BufferedReader br = null;
+    try {
+      br = Files.newBufferedReader(Paths.get(CPU_STAT_FILE), StandardCharsets.UTF_8);
+      String line;
+      while ((line = br.readLine()) != null) {
+        // looking for a line starting with "cpu<space>" which aggregates the values in all of
+        // the other "cpuN" lines.
+        if (line.startsWith("cpu ")) {
+          return line;
+        }
+      }
+    } catch (final IOException e) {
+      final String errMsg = "Failed to read cpu stat file: " + CPU_STAT_FILE;
+      logger.error(errMsg, e);
+    } finally {
+      if (br != null) {
+        try {
+          br.close();
+        } catch (final IOException e) {
+          final String errMsg = "Failed to close cpu stat file: " + CPU_STAT_FILE;
+          logger.error(errMsg, e);
+        }
+      }
+    }
+    return null;
+  }
+
+  /**
+   * Parses cpu usage information from /proc/stat file.
+   * Example of line expected with the meanings of the values below:
+   * cpu  4705  356  584    3699   23    23     0       0     0          0
+   * ---- user nice system idle iowait  irq  softirq steal guest guest_nice
+   *
+   * Method visible within the package for testing purposes.
+   *
+   * @param line the text containing cpu usage statistics
+   * @return CpuStats object. null if there is an error.
+   */
+  CpuStats getCpuStatsFromLine(final String line) {
```

Comment:
Is it a private method?

Reply:
it has package visibility for testing purposes.